arangod/Aql/IndexHint.cpp

```diff
@@ -141,10 +141,16 @@ IndexHint::IndexHint(QueryContext& query, AstNode const* node) : IndexHint() {
         } else if (name == arangodb::StaticStrings::IndexHintMaxProjections) {
           // maxProjections is a valid attribute, but handled elsewhere
           handled = true;
+        } else {
+          ExecutionPlan::invalidOptionAttribute(query, "unknown", "FOR",
+                                                name.data(), name.size());
+          handled = true;
         }
-
         if (!handled) {
-          ExecutionPlan::invalidOptionAttribute(query, "unknown", "FOR",
+          VPackBuilder builder;
+          child->getMember(0)->toVelocyPackValue(builder);
+          std::string msg = "invalid value " + builder.toString() + " in ";
```

Comment:
I'd prefer using `toJson()` here, as `toString()` will contain pretty pretting, which is not only verbose, but also less readable when contained inside another error message/warning.
```suggestion
          std::string msg = "invalid value " + builder.toJson() + " in ";
```